src/scripts/kibana-install.sh

```diff
@@ -211,6 +226,92 @@ configuration_and_plugins()
 
       log "[configuration_and_plugins] Configured encrypted communication"
     fi
+
+    # configure HTTPS communication with Elasticsearch if cert supplied and x-pack installed.
+    # Kibana x-pack installed implies it's also installed for Elasticsearch
+    if [[ -n "${HTTP_CERT}" && ${INSTALL_XPACK} -ne 0 ]]; then
+      # convert PKCS#12 certificate to PEM format
+      [ -d /etc/kibana/ssl ] || mkdir -p /etc/kibana/ssl
+      log "[configuration_and_plugins] Converting PKCS#12 archive for Elasticsearch to PEM format"
+      echo ${HTTP_CERT} | base64 -d | tee /etc/kibana/ssl/elasticsearch-http.p12
+      log "[configuration_and_plugins] Create elasticsearch-http.crt from PKCS#12 archive"
+      echo "$HTTP_CERT_PASSWORD" | openssl pkcs12 -in /etc/kibana/ssl/elasticsearch-http.p12 -out /etc/kibana/ssl/elasticsearch-http.crt -clcerts -nokeys -passin stdin
+      log "[configuration_and_plugins] Create elasticsearch-http.key from PKCS#12 archive"
+      echo "$HTTP_CERT_PASSWORD" | openssl pkcs12 -in /etc/kibana/ssl/elasticsearch-http.p12 -out /etc/kibana/ssl/elasticsearch-http.key -nocerts -nodes -passin stdin
+      log "[configuration_and_plugins] Create elasticsearch-http-ca.crt from PKCS#12 archive"
+      echo "$HTTP_CERT_PASSWORD" | openssl pkcs12 -in /etc/kibana/ssl/elasticsearch-http.p12 -out /etc/kibana/ssl/elasticsearch-http-ca.crt -cacerts -nokeys -chain -passin stdin
+      log "[configuration_and_plugins] Configuring TLS for Elasticsearch"
+      echo "elasticsearch.ssl.key: /etc/kibana/ssl/elasticsearch-http.key" >> $KIBANA_CONF
+
+      if dpkg --compare-versions "$KIBANA_VERSION" ">=" "5.3.0"; then
+        echo "elasticsearch.ssl.certificate: /etc/kibana/ssl/elasticsearch-http.crt" >> $KIBANA_CONF
+        # A user may provide a certificate that would fail full verification mode,
+        # so default to certificate mode which verifies that the provided certificate is signed
+        # by a trusted authority (CA), but does not perform any hostname verification.
```

Comment:
I think the template should take an optional CA certificate `elasticsearchCaCertificate` and use `certgen` to generate the elasticsearch node's HTTP and Transport certificates.  This way we do not have to lower the verification mode. 

Reusing the same HTTP/Transport certificate for multiple nodes is not something we should promote.

Reply:
To do that, you'd need to have the key for the CA, which is unlikely to be the case if they're using an existing CA.
For transport our recommendation is to have a cluster-specific CA. But for HTTP, there's a good chance they'd want to use their own CA so that their HTTP clients are already configured to trust it.